starlark/eval_test.go

```diff
@@ -109,13 +110,17 @@ func TestExecFile(t *testing.T) {
 		"testdata/set.star",
 		"testdata/string.star",
 		"testdata/tuple.star",
+		"testdata/recursion.star",
 	} {
 		filename := filepath.Join(testdata, file)
 		for _, chunk := range chunkedfile.Read(filename, t) {
 			predeclared := starlark.StringDict{
 				"hasfields": starlark.NewBuiltin("hasfields", newHasFields),
 				"fibonacci": fib{},
 			}
+
+			resolve.AllowRecursion = option(chunk.Source, "recursion")
```

Comment:
Ideally all the other flags would be off by default and enabled in this way too. We can do that as a followup.